lib/tools.js

```diff
@@ -68,67 +68,64 @@ function asString (str) {
 }
 
 function asJson (obj, msg, num, time) {
-  // to catch both null and undefined
-  const hasObj = obj !== undefined && obj !== null
-  const objError = hasObj && obj instanceof Error
-  msg = !msg && objError === true ? obj.message : msg || undefined
+  var objectWithMessage = Object.assign({}, obj)
```

Comment:
this is likely going to cause bottlenecks, particular for objects with a lot of top level properties

Reply:
I think, I can do w/o copying the object. This means the metadata will contain the property msg/messageKey
Is this acceptable?